### PR TITLE
Allow DVCs to view the team management page

### PR DESCRIPTION
**Pull request by hschallhorn**
bumps #14608

### Description
Allows DVCs to access the team management page, but only view judge teams.

### Acceptance Criteria
- [x] DVCs can navigate to the team management page from their main dropdown
- [x] DVCs can view the team management page
- [x] DVCs can only see judge teams on the team management page
- [x] DVCs cannot navigate to judge team management page from the caseflow team management page (no org admin page link)

### Testing Plan
1. Run `make reset` or just the following commands
   ```ruby
   dvc = User.create(css_id: "BVATCOLLIER", station_id: 101, full_name: "Tonja DVCTeam Collier")
   dvc_team = DvcTeam.create_for_dvc(dvc)
   ```
1. Log in as BVATCOLLIER
1. Click "Caseflow team management" from your main dropdown

### User Facing Changes

 BVA ADMIN|DVC
 ---|---
Screen Shot 2020-08-31 at 2 32 45 PM|Screen Shot 2020-08-31 at 2 36 06 PM

### CODE REVIEW

File: client/app/queue/TeamManagement.jsx [hunk 1]
```diff
@@ -61,28 +61,34 @@ class TeamManagement extends React.PureComponent {
...
+                    />
+                  </span>
+                </OrgHeader>
+                <OrgList orgs={this.props.privateBars} isRepresentative />
+              </React.Fragment> }
+
+              { this.props.otherOrgs && <React.Fragment>
+                <OrgHeader>Other teams</OrgHeader>
+                <OrgList orgs={this.props.otherOrgs} />
+              </React.Fragment> }
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:91]: Only show all other orgs if they are returned from the back end

File: client/app/queue/TeamManagement.jsx [hunk 2]
```diff
@@ -263,7 +269,7 @@ class OrgRow extends React.PureComponent {
         }
       </td>
       <td>
-        { this.state.url && <Link to={this.state.user_admin_path}>
+        { this.state.url && this.state.user_admin_path && <Link to={this.state.user_admin_path}>
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:272]: Only show path to team admin page if it exists

File: app/controllers/team_management_controller.rb [hunk 3]
```diff
@@ -80,7 +91,11 @@ def serialize_org(org)
       participant_id: org.participant_id,
       type: org.type,
       url: org.url,
-      user_admin_path: org.user_admin_path
+      user_admin_path: current_user.can_view_only_judge_team_management? ? nil : org.user_admin_path
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:94]: [Don't show team admin pages to DVCs](https://github.com/department-of-veterans-affairs/caseflow/pull/14928/files#diff-2c108c5dd4d1395197feef5a3465a057R272)

File: app/controllers/team_management_controller.rb [hunk 4]
```diff
@@ -80,7 +91,11 @@ def serialize_org(org)
       participant_id: org.participant_id,
       type: org.type,
       url: org.url,
-      user_admin_path: org.user_admin_path
+      user_admin_path: current_user.can_view_team_management? ? org.user_admin_path : nil
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:94]: [Don't show team admin pages to DVCs](https://github.com/department-of-veterans-affairs/caseflow/pull/14928/files#diff-2c108c5dd4d1395197feef5a3465a057R272)

File: app/controllers/team_management_controller.rb [hunk 5]
```diff
@@ -69,18 +64,31 @@ def update_params
...
   def serialize_org(org)
-    {
-      id: org.id,
-      name: org.is_a?(JudgeTeam) ? org.judge.full_name.titleize : org.name,
-      participant_id: org.participant_id,
-      type: org.type,
-      url: org.url,
-      user_admin_path: org.user_admin_path
-    }
+    org.serializable_hash.merge(user_admin_path: current_user.can_view_team_management? ? org.user_admin_path : nil)
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:86]: [Don't show team admin pages to DVCs](https://github.com/department-of-veterans-affairs/caseflow/pull/14928/files#diff-2c108c5dd4d1395197feef5a3465a057R272)

File: app/controllers/team_management_controller.rb [hunk 6]
```diff
@@ -69,18 +64,31 @@ def update_params
...
 
   def serialize_org(org)
-    {
-      id: org.id,
-      name: org.is_a?(JudgeTeam) ? org.judge.full_name.titleize : org.name,
-      participant_id: org.participant_id,
-      type: org.type,
-      url: org.url,
-      user_admin_path: org.user_admin_path
-    }
```
Comment (hschallhorn) [inline app/controllers/team_management_controller.rb:84]: Code climate! [Pulled out to org](https://github.com/department-of-veterans-affairs/caseflow/pull/14928/files#diff-e9b0bdeeb6d95d19520c1006758e2672R139)

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -136,22 +115,56 @@ def application_urls
...
+
+  def dropdown_urls
+    urls = defult_menu_items
+    urls.concat(admin_menu_items)
 
     if current_user.present?
       urls.append(title: "Sign Out", link: url_for(controller: "/sessions", action: "destroy"), border: true)
+      if ApplicationController.dependencies_faked?
+        urls.append(title: "Switch User", link: url_for(controller: "/test/users", action: "index"))
+      end
```
Comment (hschallhorn) [inline app/controllers/application_controller.rb:167]: Code climate and general reshuffling to keep like functions together

File: client/app/queue/TeamManagement.jsx [hunk 8]
```diff
@@ -62,47 +62,57 @@ class TeamManagement extends React.PureComponent {
 
           <table {...tableStyling}>
             <tbody>
```
Comment (hschallhorn) [inline client/app/queue/TeamManagement.jsx:64]: Whitespace mostly!

File: app/controllers/application_controller.rb (hunk 7 above)
Comment (ajspotts) [reply app/controllers/application_controller.rb:167]: what a nice example of array methods variety in `admin_menu_items` 🙌

**Review by ajspotts — APPROVED** [review]
Great changes! AC: ✅ , Testing Plan: smooth ⛵

**Pull request merged**